smack-core/src/main/java/org/jivesoftware/smack/packet/AbstractError.java

```diff
@@ -105,7 +113,8 @@ protected void addDescriptiveTextsAndExtensions(XmlStringBuilder xml) {
             String xmllang = entry.getKey();
             String text = entry.getValue();
             xml.halfOpenElement("text").xmlnsAttribute(textNamespace)
-                    .xmllangAttribute(xmllang).rightAngleBracket();
+                    .xmllangAttribute("".equals(xmllang) ? null : xmllang)
```

Comment:
That one is a bit trickier. While there is `XmlStringBuilder.optXmlLangAttribute(String)` the method only checks if the argument is `null`. But I think it's safe to extend the semantic of `optXmlLangAttribute()` to use `StringUtils.isNotNullOrEmpty(String)`, which would make it suitable here. And would avoid the conditional operator construct.